modules/core/src/smithy4s/RefinementProvider.scala

```diff
@@ -202,9 +202,14 @@ private[smithy4s] trait LowPriorityImplicits {
       : RefinementProvider[Pattern, E, E] =
     new RefinementProvider.PatternConstraint[E](e => e.value)
 
-  implicit def isomorphismConstraint[C, A, A0](implicit
+  @deprecated("Use isomorphismConstraint2 instead", "0.18.25")
```

Comment:
**thought:** for bincompat, we have to keep the old method.

Reply:
I tried to make it `private[smithy4s]` but MiMa complained about the static forwarder disappearing. Figured I'd restore the method to the public scope, it doesn't hurt if it's not implicit anymore.